### ISSUE TITLE
wry system_tray and system_tray_no_menu panic exit on windows 10

### ISSUE THREAD

**Issue by davehorner** [tauri-apps/wry#463]
**Describe the bug**
Running the system_tray examples, the system tray shows up; when clicked, a window flashes and the application exits with console errors.  Other examples are working fine.

**Steps To Reproduce**
> cargo run --example system_tray
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
     Running `target\debug\examples\system_tray.exe`
Wry has started!
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src\webview\webview2\mod.rs:235:16
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
webview2: panic in callback function. Aborting because it's UB to unwind across FFI boundaries.
error: process didn't exit successfully: `target\debug\examples\system_tray.exe` (exit code: 255)

> cargo run --example system_tray_no_menu
    Finished dev [unoptimized + debuginfo] target(s) in 3.95s
     Running `target\debug\examples\system_tray_no_menu.exe`
Wry has started!
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src\webview\webview2\mod.rs:235:16
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
webview2: panic in callback function. Aborting because it's UB to unwind across FFI boundaries.
error: process didn't exit successfully: `target\debug\examples\system_tray_no_menu.exe` (exit code: 255)

**Expected behavior**
both programs exit with a panic.  expect it to work and keep running.

**Platform and Versions (please complete the following information):**
OS: windows 10
Rustc:  rustc --version
rustc 1.56.1 (59eed8a2a 2021-11-01)

**Would you assign yourself to resolve this bug?**
- [ ] Yes
- [ X] No - do not know how.

**Additional context**
The other examples like multi_window work fine so I know that the webview2 component is working

**Comment by amrbashir** [comment]
you rust is outdated, can you try upgrading to 1.57 ? we had some reports about bugs with 1.56 but got fixed in 1.57

**Comment by davehorner** [comment]
>rustc --version
rustc 1.57.0 (f1edd0429 2021-11-29)

the behavior is the same.  the window may stay open a little longer.  The no_menu displayed a window that resized but only painted black in the newly surfaced window portions.

**Comment by amrbashir** [comment]
It is working fine on my end, can you try getting a backtrace ?

**Comment by davehorner** [comment]
```Wry has started!
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src\webview\webview2\mod.rs:235:16
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\panicking.rs:517
   1: core::panicking::panic_fmt
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\core\src\panicking.rs:100
   2: core::panicking::panic
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\core\src\panicking.rs:50
   3: enum$<core::option::Option<ref$<tuple$<alloc::string::String,alloc::boxed::Box<dyn$<core::ops::function::Fn<tuple$<ref$<wry::shared::http::request::Request> >,assoc$<Output,enum$<core::result::Result<wry::shared::http::response::Response,enum$<wry::Error>
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\core\src\option.rs:746
   4: wry::webview::webview2::impl$0::new::closure$0::closure$0::closure$4
             at .\src\webview\webview2\mod.rs:232
   5: webview2::impl$4::add_web_resource_requested::closure$0<wry::webview::webview2::impl$0::new::closure$0::closure$0::closure$4>
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\alloc\src\boxed.rs:1705
   7: webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\webview2-0.1.1\src\lib.rs:96
   8: core::ops::function::FnOnce::call_once<webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0,tuple$<> >
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\core\src\ops\function.rs:227
   9: core::panic::unwind_safe::impl$23::call_once<i32,webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0>
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\core\src\panic\unwind_safe.rs:271
  10: std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0>,i32>
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\std\src\panicking.rs:403
 11: std::panicking::begin_panic::impl$1::take_box<str>
  12: std::panicking::try<i32,core::panic::unwind_safe::AssertUnwindSafe<webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0> >       
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\std\src\panicking.rs:367
  13: std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<webview2::impl$4::add_web_resource_requested::impl$0::invoke::closure$0>,i32>   
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\std\src\panic.rs:133
  14: webview2::impl$4::add_web_resource_requested::impl$0::invoke
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\webview2-0.1.1\src\lib.rs:95
  15: webview2_sys::impl$284::vtable::icorewebview2webresourcerequestedeventhandler_invoke<webview2::impl$4::add_web_resource_requested::Impl,com::offset::Zero>
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\webview2-sys-0.1.0\src\lib.rs:2813
  16: CreateWebViewEnvironmentWithOptionsInternal
  17: CreateWebViewEnvironmentWithOptionsInternal
  18: CreateWebViewEnvironmentWithOptionsInternal
  19: CreateWebViewEnvironmentWithOptionsInternal
  20: DllCanUnloadNow
  21: CreateWebViewEnvironmentWithOptionsInternal
  22: DllCanUnloadNow
  23: DllCanUnloadNow
  24: DllCanUnloadNow
  25: DllCanUnloadNow
  26: DllCanUnloadNow
  27: DllCanUnloadNow
  28: DllCanUnloadNow
  29: DllCanUnloadNow
  30: DllCanUnloadNow
  31: DllCanUnloadNow
  32: DllCanUnloadNow
  33: Ordinal0
  34: DllCanUnloadNow
  35: CallWindowProcW
  36: DispatchMessageW
  37: tao::platform_impl::platform::event_loop::EventLoop<tuple$<> >::run_return<tuple$<>,system_tray::main::closure$0>
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\tao-0.5.2\src\platform_impl\windows\event_loop.rs:248
  38: tao::platform_impl::platform::event_loop::EventLoop<tuple$<> >::run<tuple$<>,system_tray::main::closure$0>
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\tao-0.5.2\src\platform_impl\windows\event_loop.rs:203
  39: tao::event_loop::EventLoop<tuple$<> >::run<tuple$<>,system_tray::main::closure$0>
             at C:\Users\dhorner\.cargo\registry\src\github.com-1ecc6299db9ec823\tao-0.5.2\src\event_loop.rs:151
  40: system_tray::main
             at .\examples\system_tray.rs:104
  41: core::ops::function::FnOnce::call_once<enum$<core::result::Result<tuple$<>,enum$<wry::Error> >, 0, 17, Err> (*)(),tuple$<> >
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\core\src\ops\function.rs:227
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
webview2: panic in callback function. Aborting because it's UB to unwind across FFI boundaries.```

**Comment by amrbashir** [comment]
I think you are using `dev` branch, can you try the `next` branch, we have switched from `webview2` crate to `webview2-com`

**Comment by davehorner** [comment]
I just pulled from the git repo and the default branch was next.  It builds and runs correctly.

I do see that the webview2 crate has been replaced with webview2-com.  I did a directory comparison and both Cargo.toml files claim to be 0.12.2; did the change in dependency not come with a version increment of some sort?

The Cargo.toml that I started with indicates it is: # THIS FILE IS AUTOMATICALLY GENERATED BY CARGO

I pulled down the code for wry locally using 'cargo clone wry'.  Trying to understand and hopefully keep it from happening again to others.
